node/router/tests/disconnect.rs

```diff
@@ -33,8 +34,12 @@ async fn test_disconnect_without_handshake() {
 
     // Connect node0 to node1.
     node0.connect(node1.local_ip());
-    // Sleep briefly.
-    tokio::time::sleep(Duration::from_millis(200)).await;
+    // Await both nodes being connected.
+    let node0_ = node0.clone();
+    let node1_ = node1.clone();
+    deadline!(Duration::from_secs(5), move || {
```

Comment:
If these test cases usually run in milliseconds, we can probably lower the upper bound here a little?

Reply:
Thanks! I've reduced it to 1 second.